v-next/hardhat-ignition-core/src/types/artifact.ts

```diff
@@ -29,6 +42,9 @@ export interface Artifact<AbiT extends Abi = Abi> {
   deployedBytecode: string;
   linkReferences: LinkReferences;
   deployedLinkReferences: LinkReferences;
+  immutableReferences?: ImmutableReferences;
+  buildInfoId?: string;
+  inputSourceName?: string;
```

Comment:
Can I confirm my understanding here.
These properties are new in HH3. We represent them as optional so that we can represent both versions of the artifact with one type?
What do you think of having two separate types and detecting during verify/migrate, and going with the HH3 type in all other usage?

Reply:
The only reason these are optional is because they are optional in the Hardhat type as well. That being said, I'm not opposed to having two separate types if we decide to go that direction.